internal/dag/builder.go

```diff
@@ -787,6 +807,38 @@ func (b *Builder) computeRoutes(sw *ObjectStatusWriter, proxy *projcontour.HTTPP
 	return routes
 }
 
+func validateHeaderAlterations(alt *projcontour.HeaderAlterations) (map[string]string, []string, error) {
+	if alt == nil {
+		return nil, nil, nil
+	}
+	add := make(map[string]string)
+
+	for _, entry := range alt.Add {
```

Comment:
Should we handle headers with the `:` prefix explicit, or just let `validation.IsHTTPHeaderName` deal with it? I'm inclined towards the latter ...